src/DI/WebpackExtension.php

```diff
@@ -110,4 +114,44 @@ public function beforeCompile()
 		}
 	}
 
+
+	private function setupAssetResolver(array $config): ServiceDefinition
+	{
+		$builder = $this->getContainerBuilder();
+
+		$assetResolver = $builder->addDefinition($this->prefix('assetResolver'))
+			->setClass(AssetResolver\AssetResolverInterface::class);
+
+		if ($config['manifest']['name'] !== NULL) {
+			Validators::assertField($config['manifest'], 'name', 'string:1..');
+			$loader = $builder->addDefinition($this->prefix('manifestLoader'))
+				->setClass(ManifestLoader::class);
+
+			$args = [
+				$config['manifest']['name'],
+				$loader,
+			];
+
+			if ($config['manifest']['optimize']) {
+				$devServerInstance = new DevServer($config['devServer']['enabled'], $config['devServer']['url'], new Client());
+				$directoryProviderInstance = new BuildDirectoryProvider($config['build']['directory'], $devServerInstance);
```

Comment:
BuildDirectoryProvider, there is something wrong about this class. It has only one condition in it. It looks really too timy for me. Maybe it can provide full path to given file, it seems more natural to me. And and name... AssetLocator? BundleLocator? ArtifactsLocator?